### PR TITLE
Set titles to current page header

### LINKED ISSUE #419: `${h1Content} - Linode Manager` title scheme
In the login service, the page `<title>` scheme has been to use the title of the page (indicated by the text within `<h1>` on the page) as the first part of the `<title>` followed by "- Linode Manager". This is a common scheme and has the benefit of showing the most important information first (rather than the less important "what site am I on"). We should use the same scheme in the Manager especially now that the two services are so integrated.

### PR DESCRIPTION
**Pull request by briansteffens**
Closes #419

### CODE REVIEW

File: src/reducers/title.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { SET_TITLE } from '~/actions/title';
+
+export default function title(state = null, action) {
+  switch (action.type) {
+    case SET_TITLE:
+      document.title = `${action.title} - Linode Manager`;
```
Comment (mparke) [inline src/reducers/title.js]: We should take a look at using the existing pattern, `{BrandName} - {PageTitle}`, unless we have a good reason for changing?

For example, 

`/` == `Linode - Linode Manager`
`/dnsmanager` == `Linode - DNS Manager`
etc

Prefix could be defined here at the reducer level, ( Like the suffix "Linode Manager" is now ).

**Comment by mparke** [comment]
I think we're missing pages in this pull.

File: src/reducers/title.js (hunk 1 above)
Comment (eatonphil) [inline src/reducers/title.js]: This is not a good convention because it hides the page's useful information first.

**Comment by briansteffens** [comment]
I could modify the stubs for longview and nodebalancers so they have a componentDidMount and set the titles for those but it didn't seem important since they're basically just TODOs at the moment. Any other missed pages?

File: src/reducers/title.js (hunk 1 above)
Comment (mparke) [inline src/reducers/title.js]: Hmm, I hear you. Do we also need to consider cases where you're navigating within a Linode (or for the sake of argument, some object)?

I ask, based on an existing, for example:

```
Linode - Dashboard // my_linode_name
```

What's nice about this pattern is it acts more closely to a breadcrumb than the inverted version would. By inverted, I mean, with a suffix as defined here, 

```Dashboard - Linode Manager```

Dashboard for which Linode?

```
Dashboard - my_linode_name - Linode Manager
```

I'd argue it's easier to follow if the title follows the same pattern as the url

```
Brand - Page
Brand - Object - Page
```

Just wanted to bring up options. Let me know what you guys think

**Comment by mparke** [comment]
@briansteffens We should probably make sure titles are defined for every page that has unique content/purpose. This applies to most tabs within a Linode for example (alpha manager). The existing manager has solid examples

File: src/reducers/title.js (hunk 1 above)
Comment (briansteffens) [inline src/reducers/title.js]: I would agree except that the title in a tab gets cut off if it's too long, so I think titles being inverted to show the most specific information first makes sense, it's more likely that the interesting information will not be part of the subset which gets cut off.

**Comment by briansteffens** [comment]
Hmm I could see it either way.

**Comment by briansteffens** [comment]
So like:
`Rebuild - {label} - Linode Manager`

What about the edit config page?
`{config label} - {label} - Linode Manager`
or
`{config label} - Configs - Settings - {label} - Linode Manager`

Is there a pattern that applies to all pages?

**Comment by eatonphil** [comment]
The title won't even show up for more than 30 characters. I think we should just go with the h1 title of the current page and the Linode brandname to keep things concise.

**Comment by briansteffens** [comment]
Oh now I remember the discussion about not having a 'Linode Manager', just linode.com - so we should drop the 'Manager' part too.

**Comment by mparke** [comment]
sgtm re: concise for now.

**Comment by briansteffens** [comment]
Okay after recent commit it's down to `{label} - Linode` good?

**Pull request merged**